app/models/llanfair.rb

```diff
@@ -0,0 +1,113 @@
+module Llanfair
+  class Run < Run
+    def self.sti_name
+      :llanfair
+    end
+  end
+
+  class Split < Split
+  end
+
+  class Parser < BabelBridge::Parser
+    def parse(character_array)
+      # Convert the array of character from string form to an actual array
+      # Then pack the characters back to the original bytes
+      file = RunFile.pack_binary(character_array)
+      run = {}
+      run[:time] = 0
+      # A port of Nitrofski's import llanfair files function for wsplit
+      file = StringIO.new(file)
+      file.seek(197, 1)
+      goal_len = file.read(2).unpack("n")[0]
+      run[:goal] = file.read(goal_len)
```

Comment:
Hmm, it's just now hit me that Llanfair has both a Title and a Goal. What's the typical use of these fields for Llanfair users? I imagine Goal is mostly "`gamename` `categoryname`"? Or is Title typically `gamename` and Goal is typically `categoryname`?

The three standard fields for inputs we use are `run[:game]`, `run[:category]`, and `run[:name]` (with `run[:name]` defaulted to "`run[:game]` `run[:category]`" for some timers), so it's worth thinking about if the conventional uses for Title or Goal fit into two of those three slots.